src/iks003_incompressibleLinearElasticity.cpp

```diff
@@ -15,84 +15,86 @@
 #include <ikarus/utils/drawing/griddrawer.hh>
-#include <ikarus/utils/duneUtilities.hh>
-#include <ikarus/utils/eigenDuneTransformations.hh>
 #include <ikarus/utils/init.hh>
 
 using namespace Ikarus;
 using namespace Dune::Indices;
-template <typename Basis>
-struct Solid : public Ikarus::AutoDiffFE<Solid<Basis>, typename Basis::FlatBasis> {
-  using BaseAD = Ikarus::AutoDiffFE<Solid<Basis>, typename Basis::FlatBasis>;
-  friend BaseAD;
-  using BaseAD::localView;
+template <typename Basis_, typename FERequirements_ = FErequirements<>, bool useEigenRef = false>
+struct Solid {
+ public:
+  using Basis             = Basis_;
   using FlatBasis         = typename Basis::FlatBasis;
   using LocalView         = typename FlatBasis::LocalView;
-  using FERequirementType = typename BaseAD::FERequirementType;
-  using Traits            = TraitsFromLocalView<LocalView>;
+  using Element           = typename LocalView::Element;
+  using Geometry          = typename Element::Geometry;
+  using FERequirementType = FERequirements_;
+  using Traits            = TraitsFromLocalView<LocalView, useEigenRef>;
   Solid(const Basis &basis, const typename LocalView::Element &element, double emod, double nu)
-      : BaseAD(basis.flat(), element), emod_{emod}, nu_{nu} {
-    this->localView().bind(element);
+      : localView_{basis.flat().localView()}, emod_{emod}, nu_{nu} {
+    localView_.bind(element);
     mu_       = emod_ / (2 * (1 + nu_));
     lambdaMat = convertLameConstants({.emodul = emod_, .nu = nu_}).toLamesFirstParameter();
   }
 
   using GlobalIndex = typename LocalView::MultiIndex;
   void globalFlatIndices(std::vector<GlobalIndex> &globalIndices) const {
-    const auto &feDisp = this->localView().tree().child(_0, 0).finiteElement();
+    const auto &feDisp = localView_.tree().child(_0, 0).finiteElement();
     for (size_t i = 0; i < feDisp.size(); ++i) {
       for (int j = 0; j < Traits::worlddim; ++j) {
-        globalIndices.push_back(this->localView().index((this->localView().tree().child(_0, j).localIndex(i))));
+        globalIndices.push_back(localView_.index((localView_.tree().child(_0, j).localIndex(i))));
       }
     }
-    const auto &fePressure = this->localView().tree().child(_1).finiteElement();
+    const auto &fePressure = localView_.tree().child(_1).finiteElement();
     for (size_t i = 0; i < fePressure.size(); ++i) {
-      globalIndices.push_back(this->localView().index((this->localView().tree().child(_1).localIndex(i))));
+      globalIndices.push_back(localView_.index((localView_.tree().child(_1).localIndex(i))));
     }
   }
 
- private:
+  [[nodiscard]] constexpr int size() const { return localView_.size(); }
+  const Element &gridElement() { return localView_.element(); }
+  const LocalView &localView() const { return localView_; }
+  LocalView &localView() { return localView_; }
+
+ protected:
   template <class ScalarType>
   [[nodiscard]] ScalarType calculateScalarImpl(const FERequirementType &par,
```

Comment:
I think the calculateScalar implementation should also follow the other code. Can you refactor with `std::optional`? Then you doint have to provide always the argoment